.github/workflows/test_packages.yml

```diff
@@ -52,10 +45,27 @@ jobs:
         id: configure
         run: python ./build_tools/github_actions/fetch_test_configurations.py
 
+  test_sanity_check:
+    name: 'Test Sanity Check'
+    needs: configure_test_matrix
+    uses: './.github/workflows/test_sanity_check.yml'
+    with:
+      artifact_run_id: ${{ inputs.artifact_run_id }}
+      amdgpu_families: ${{ inputs.amdgpu_families }}
+      test_runs_on: ${{ inputs.test_runs_on }}
+      platform: ${{ needs.configure_test_matrix.outputs.platform }}
+
   test_components:
     name: 'Test ${{ matrix.components.job_name }}'
     runs-on: ${{ inputs.test_runs_on }}
-    needs: configure_test_matrix
+    container:
+      image: ${{ needs.configure_test_matrix.outputs.platform == 'linux' && 'ghcr.io/rocm/no_rocm_image_ubuntu24_04:main' || null }}
```

Comment:
Let's pin the image to a hash here so we can reproduce workflow results after changing the images, roll back to known working images after changing them, etc.

Similar to how we do it here: https://github.com/ROCm/TheRock/blob/ec1c2ef4f2636bce7733fd8c95e1dbb6692c8a57/.github/workflows/build_portable_linux_pytorch_wheels.yml#L84-L85